### PR TITLE
[16.0] session_db

### PR DESCRIPTION
**Pull request by sbidoul**
Migrate `session_db` from https://github.com/odoo/odoo-extra/tree/master/session_db to 16.0.

I'm going to play with this and see if it fares well enough, so I keep the PR draft for now.

@nseinlet are you ok with us maintaining your session_db module in OCA? As there was no license in odoo-extra, I chose LGPL. Let me know if there is any concern with that.

### CODE REVIEW

File: session_db/session.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        self._cr.execute("DELETE FROM http_sessions WHERE sid=%s", [session.sid])
+
+    @with_cursor
+    def get(self, sid):
+        self._cr.execute(
+            "UPDATE http_sessions "
+            "SET write_date = now() at time zone 'UTC' "
+            "WHERE sid=%s",
+            [sid],
+        )
```
Comment (sbidoul) [inline session_db/session.py]: Updating the write_date when reading is likely not necessary. If I read the code correctly the FilesystemSessionStore does not do that.

File: session_db/session.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+    # This method is not part of the Session interface but is called nevertheless,
+    # so let's get it from FilesystemSessionStore.
+    rotate = http.FilesystemSessionStore.rotate
+
+    @with_cursor
+    def vacuum(self):
+        self._cr.execute(
+            "DELETE FROM http_sessions "
+            "WHERE now() at time zone 'UTC' - write_date > '7 days'"
```
Comment (sbidoul) [inline session_db/session.py]: use SESSION_LIFETIME constant (which is now 90 days, btw)

File: session_db/session.py (hunk 1 above)
Comment (nseinlet) [reply session_db/session.py]: The point of doing this is to have the session not expire after 8 days.

**Comment by nseinlet** [comment]
Yes, nice to see it useful

File: session_db/session.py (hunk 1 above)
Comment (sbidoul) [reply session_db/session.py]: Yes, but the FilesystemSessionStore does not do that right? Any specific reason to do it differently here?

**Review by adrienpeiffer — APPROVED** [review]

**Review by lmignon — APPROVED** [review]
LGTM (Code + functional)

**Review by AnizR — APPROVED** [review]
Code LGTM, do you plan to make PGSessionStore thread safe?

**Comment by sbidoul** [comment]
> Code LGTM, do you plan to make PGSessionStore thread safe?

No, this is really meant to be used with multiple workers.

I'm going ahead and merging this. I'll remain available in case any issue show up.

/ocabot merge patch

**Pull request merged**